demo/theme-vaadin-input-md.html

```diff
@@ -0,0 +1,39 @@
+<dom-module id="theme-vaadin-input-md">
+  <template>
+    <style>
+      html {
+        --vaadin-input-input: {
+          outline: none;
+          border: none;
+          background: transparent no-repeat;
+          background-image:
+           linear-gradient(to bottom, var(--primary-color, deepskyblue), var(--primary-color, deepskyblue)),
+           linear-gradient(to bottom, var(--secondary-text-color, silver), var(--secondary-text-color, silver));
+          background-size: 0 2px, 100% 1px;
+          background-position: 50% 100%, 50% 100%;
+
+          font-family: sans-serif;
+          font-size: 16px;
+          @apply --paper-font-subhead;
+        };
+
+        --vaadin-input-input-focus: {
+          transition: background-size 0.3s ease-out;
```

Comment:
As a best practice, only transition opacity or transforms. That way those get hardware accelerated and don’t cause a repaint for every frame of the animation.

So, it is better to use an actual element (pseudo element in this case) for the underline, not background color. Although, I like the idea otherwise, but because of the animation/transition, we should not use it.